src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -113,40 +114,11 @@ public AbstractEntityAIGuard(@NotNull final J job)
-        leatherLevelArmor.add(new GuardItems(ToolType.HELMET, EntityEquipmentSlot.HEAD,  ARMOR_LEVEL_LEATHER, 1, 0, 20, 1, 3));
-        leatherLevelArmor.add(new GuardItems(ToolType.LEGGINGS, EntityEquipmentSlot.LEGS,  ARMOR_LEVEL_LEATHER, 1, 0, 20, 1, 3));
-        itemsNeeded.add(leatherLevelArmor);
-
-        final List<GuardItems> goldLevelArmor = new ArrayList<>();
-        goldLevelArmor.add(new GuardItems(ToolType.BOOTS, EntityEquipmentSlot.FEET,  ARMOR_LEVEL_GOLD, 1, 0, 20, 1, 4));
-        goldLevelArmor.add(new GuardItems(ToolType.CHESTPLATE, EntityEquipmentSlot.CHEST, ARMOR_LEVEL_GOLD, 1, 0, 20, 1, 4));
-        goldLevelArmor.add(new GuardItems(ToolType.HELMET, EntityEquipmentSlot.HEAD,  ARMOR_LEVEL_GOLD, 1, 0, 20, 1, 4));
-        goldLevelArmor.add(new GuardItems(ToolType.LEGGINGS, EntityEquipmentSlot.LEGS,  ARMOR_LEVEL_GOLD, 1, 0, 20, 1, 4));
-        itemsNeeded.add(goldLevelArmor);
-
-        final List<GuardItems> chainLevelArmor = new ArrayList<>();
-        chainLevelArmor.add(new GuardItems(ToolType.BOOTS, EntityEquipmentSlot.FEET, ARMOR_LEVEL_CHAIN, 1, 0, 20, 2, 5));
-        chainLevelArmor.add(new GuardItems(ToolType.CHESTPLATE, EntityEquipmentSlot.CHEST, ARMOR_LEVEL_CHAIN, 1, 0, 20, 2, 5));
-        chainLevelArmor.add(new GuardItems(ToolType.HELMET, EntityEquipmentSlot.HEAD, ARMOR_LEVEL_CHAIN, 1, 0, 20, 2, 5));
-        chainLevelArmor.add(new GuardItems(ToolType.LEGGINGS, EntityEquipmentSlot.LEGS, ARMOR_LEVEL_CHAIN, 1, 0, 20, 2, 5));
-        itemsNeeded.add(chainLevelArmor);
-
-        final List<GuardItems> ironLevelArmor = new ArrayList<>();
-        ironLevelArmor.add(new GuardItems(ToolType.BOOTS, EntityEquipmentSlot.FEET, ARMOR_LEVEL_IRON, 1, 5, 30, 3, 5));
-        ironLevelArmor.add(new GuardItems(ToolType.CHESTPLATE, EntityEquipmentSlot.CHEST, ARMOR_LEVEL_IRON, 0, 5, 30, 3, 5));
-        ironLevelArmor.add(new GuardItems(ToolType.HELMET, EntityEquipmentSlot.HEAD, ARMOR_LEVEL_IRON, 1, 5, 30, 3, 5));
-        ironLevelArmor.add(new GuardItems(ToolType.LEGGINGS, EntityEquipmentSlot.LEGS, ARMOR_LEVEL_IRON, 1, 5, 30, 3, 5));
-        itemsNeeded.add(ironLevelArmor);
-
-        final List<GuardItems> diamondLevelArmor = new ArrayList<>();
-        diamondLevelArmor.add(new GuardItems(ToolType.BOOTS, EntityEquipmentSlot.FEET, ARMOR_LEVEL_DIAMOND, 1, 15, 99, 4, 5));
-        diamondLevelArmor.add(new GuardItems(ToolType.CHESTPLATE, EntityEquipmentSlot.CHEST, ARMOR_LEVEL_DIAMOND, 1, 15, 99, 4, 5));
-        diamondLevelArmor.add(new GuardItems(ToolType.HELMET, EntityEquipmentSlot.HEAD, ARMOR_LEVEL_DIAMOND, 1, 15, 99, 4, 5));
-        diamondLevelArmor.add(new GuardItems(ToolType.LEGGINGS, EntityEquipmentSlot.LEGS, ARMOR_LEVEL_DIAMOND, 1, 15, 99, 4, 5));
-        itemsNeeded.add(diamondLevelArmor);
+        itemsNeeded.add(GuardGearBuilder.buildGearForLevel(ARMOR_LEVEL_LEATHER, LEATHER_LEVEL_RANGE, LEATHER_BUILDING_LEVEL_RANGE));
```

Comment:
Even though it is not what I mean with a Builder class, this makes it way more readable too.